src/main/java/emissary/util/TimeUtil.java

```diff
@@ -232,6 +234,34 @@ public static ZonedDateTime getZonedDateFromISO8601(@Nullable final String dateS
         return dateString == null ? null : ZonedDateTime.parse(dateString, DATE_ISO_8601);
     }
 
+    /**
+     * Parses hex date string into formatted string ("yyyy-MM-dd HH:mm:ss.SSS")
+     *
+     * @param hexDate the string representation of the date
+     * @return Converted time in string format yyyy-MM-dd HH:mm:ss.SSS
+     * @throws DateTimeParseException if could not parse date
+     *
+     */
+    public static String convertHexDate(String hexDate) {
+        String hexNumber = hexDate.substring(2);
+        String dateHex = hexNumber.substring(0, 8);
+        String timeHex = hexNumber.substring(8, 16);
+
+        long daysToAdd = Long.parseLong(dateHex, 16);
+        long millisToAdd = Math.round(Long.parseLong(timeHex, 16) * 10 / 3.0);
+
+        DateTimeFormatter dtf = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss.SSS");
+
+        String startingDate = "1900-01-01 00:00:00.000";
+        LocalDateTime ldt;
+
+        ldt = LocalDateTime.parse(startingDate, dtf);
```

Comment:
What happened to the try around this?

Reply:
I initially removed the try block since enhanceHexDate no longer had access to the logger. Although now that enhanceHexDate is a standalone function, it may be more appropriate to throw an exception rather than return current date. 
To preserve the legacy behavior, I added another overloaded method that returns a default string if parse exception is thrown. This can be called as `TimeUtil.convertHexDate(dateString,LocalDateTime.now().toString())`.

Reply:
Let's give TimeUtil its own Logger and use that.

Reply:
I'm in favor of the logger additions. 

I'd also be in favor of marking the new "legacy" method with an `@Deprecated` tag so that its obvious we prefer not to use this method and move away from it in the future. @drivenflywheel does that help to alleviate some of your concerns?